### ISSUE TITLE
All modules and receivers will have to be updated first, because the protocol between the Radio and the Module has been updated

### ISSUE THREAD

**Issue by 4qr1hu9n7** [FrSkyRC/ETHOS-Feedback-Community#4734]
When 1.6.0 is released. This update. Does it include legacy receivers?

**Comment by RealTadango** [comment]
Weird "release" note. Why do i have to update? Everything works fine now so why is an update for the RF part needed for an Ethos update?

**Comment by strgaltdel** [comment]
i do understand that Ethos 1.6 is mandatory for RF  Firmware V3
https://github.com/FrSkyRC/RF_Feedback_Group/releases

but this note suggests that even for Accst & access (classic Archer)  receivers you would have to update both: Tx & Rx
i cann't believe that's meant in such a way

(imagine the user echo if these RX's wouldn’t get a firmware update to be supported under Ethos 1.6)

**Comment by RealTadango** [comment]
But is an RF update mandatory for 1.6? I hope not but the text suggest it...

**Comment by ETHOSWarrier** [comment]
I cant see any firmware for ACCST receivers or even  ACCESS Archer receivers,only Archer Plus seems to get Beta V3...Does this mean i can't  use my ACCST rx or older NON Archer plus, Archer rx.     Very worrying..EW

**Comment by 4qr1hu9n7** [comment]
I have a dozen Sr-10 Pro receivers. Will this update impact these legacy receivers? Also. Is this release a "mandatory" update or optional? Currently. All of my equipment functions flawlessly. I would hate to upset all of the hard work that went into programming and setting up of all of my models!

**Comment by Alex00101** [comment]
I can also see that for - at least - Archer Stabs - it involves yet another factory reset and the full stab. setup including calibration (i.e. taking the RC out of the model). This is just after we had to do it for the previous update (when Scripts V3 were released) - I do not find this an acceptable practice.

**Comment by bsongis-frsky** [comment]
Sorry. I wrote this note a little bit too quickly. This is needed for the new "Rx VFR" telemetry value, which is available on dual band systems

**Comment by 4qr1hu9n7** [comment]
> I can also see that for - at least - Archer Stabs - it involves yet another factory reset and the full stab. setup including calibration (i.e. taking the RC out of the model). This is just after we had to do it for the previous update (when Scripts V3 were released) - I do not find this an acceptable practice.

OH. You are so right! There is no way I'm going to tear out my receivers to recalibrate the receiver to the latest firmware release!! That's completely and totally unacceptable given the effort that went into the initial calibration and installation !!!

**Comment by 4qr1hu9n7** [comment]
Wait a minute!! This issue is not closed!! Who are you to close such an important topic? How about answering my question.

Is this release a "mandatory" update or optional? Currently.

**Comment by Alex00101** [comment]
> Wait a minute!! This issue is not closed!! Who are you to close such an important topic? How about answering my question.
> 
> Is this release a "mandatory" update or optional? Currently.

Calm down. Bertrand is Ethos lead developer and this is his prerogative. 
The issue I'm concerned about is not Ethos but FrSky RX firmware,  I'm going to raise it in their relevant gothub.

**Comment by 4qr1hu9n7** [comment]
I think that there will be a great deal of people that will no longer trust the latest release issued. I'm on 1.5.15. It works very well. That's to say I've not come across any issues that affect my experience. There was a rf module update that corrupted my telemetry and cause phantom alarms. Whatever the update was it upset my transmitter. Thus, I rolled it back to the previous working firmware. These issues are becoming more and more prevalent in each release update. I've come to a conclusion that 1.5.15 is my final update until there is a "critical" release that addresses a "safety" issue. Calm down. Yourself!

**Comment by ETHOSWarrier** [comment]
Agreed,we are grateful for the development and passion for improvement on ETHOS but I am slightly confused over this V3 release because it says it corrects(in conjunction with 1.6.0 i understand) the failsafe issues by saving the data to the rx for permanent storage and not just waiting for 10/15 sec to receive this data every time we switch on.... So the point is ,as I see it..if this is a failsafe correction,and its not mandatory then that's a total contradiction in terms. If the failsafe issue is bad enough for a re write then surely it should be mandatory,and in which case if i choose not to update to v3 and ETHOS 1.6.0 I am missing out on a safety update.   I don't want to update all my rx and tx ,just for a small failsafe issue.If the failsafe issue is solved in 1.6.0 ETHOS alone without the V3 rf upgrade,thats fine as its a developement of ETHOS.....I ,like many on here are trying to find a safe version and place to STOP any upgrades and enjoy flying but it never comes....A lot of the fixes i see i would need for a stable ETHOS are in 1.6.0 so its totally frustrating. I would love all the fixes to be addressed in one release(like futaba would do) so i get a bug free ETHOS on which to STOP and fly with for the next 15yrs if i don't need any more Features. There are bug fixes in the future releases so I cant stop now as i might hit on a bug in the future if trying something different and all the requests for new features will create more bugs....Its endless and frustrating...I am sold on Frsky and love the radios but crave a safe and stable setup for my planes so I can enjoy flying one again with no issues..EW

**Comment by RealTadango** [comment]
I will not update the RF for at least a few months after a release and if that means i cannot update to Ethos 1.6.0 so be it. For me this message only means that if you want all new features you must update to the new RF also. And that will not happen any time soon in my planes. FrSky has been "expirimental" with all RF updates lately (Why no mention of the 1.1.14 pull back? How dangerous is this release?) and WILL reset a lot of settings so i am only updating after months of testing by others ;) To much work for me for to update all planes......

**Comment by strgaltdel** [comment]
... when in did read the new note today in the morning my worries were gone:
i'll update to 1.6(n) and stay with current RF firmware for a while
no need to get upset
time will reveal when it is the right moment for an RF update.

[Ethos 1.6.0 nightlies](https://github.com/FrSkyRC/ETHOS-Feedback-Community/releases/tag/nightly16) Pre-release
Coming soon!

**The internal RF module and the TD / TW / AP / AP Plus receivers have to be upgraded if you need some of the improvements below, because the protocol between the Radio and the Module has been updated. These are the new functions which have been added** : 

**New "Rx VFR" telemetry value
The failsafe data is saved on the receiver
All stab functions have been improved
Other receivers don't need any upgrade.**

**Comment by ETHOSWarrier** [comment]
> ... when in did read the new note today in the morning my worries were gone: i'll update to 1.6(n) and stay with current RF firmware for a while no need to get upset time will reveal when it is the right moment for an RF update.
> 
> [Ethos 1.6.0 nightlies](https://github.com/FrSkyRC/ETHOS-Feedback-Community/releases/tag/nightly16) Pre-release Coming soon!
> 
> **The internal RF module and the TD / TW / AP / AP Plus receivers have to be upgraded if you need some of the improvements below, because the protocol between the Radio and the Module has been updated. These are the new functions which have been added** :
> 
> **New "Rx VFR" telemetry value The failsafe data is saved on the receiver All stab functions have been improved Other receivers don't need any upgrade.**

Yes,but thats my point ,if you do that you don't get the failsafe amendment as you need V3 in conjunction with 1.6.0 for that to work.This is the contradiction.I have already rolled back to 1.1.13 on my x20r from a buggy 1.1.14 and had to find that out my self,good job it was at home before flying and not at the field...The rf side is serious and I have got a bit twitchy now on future updates as well as others have...EW

**Comment by RealTadango** [comment]
> > ... when in did read the new note today in the morning my worries were gone: i'll update to 1.6(n) and stay with current RF firmware for a while no need to get upset time will reveal when it is the right moment for an RF update.
> > [Ethos 1.6.0 nightlies](https://github.com/FrSkyRC/ETHOS-Feedback-Community/releases/tag/nightly16) Pre-release Coming soon!
> > **The internal RF module and the TD / TW / AP / AP Plus receivers have to be upgraded if you need some of the improvements below, because the protocol between the Radio and the Module has been updated. These are the new functions which have been added** :
> > **New "Rx VFR" telemetry value The failsafe data is saved on the receiver All stab functions have been improved Other receivers don't need any upgrade.**
> 
> Yes,but thats my point ,if you do that you don't get the failsafe amendment as you need V3 in conjunction with 1.6.0 for that to work.This is the contradiction.I have already rolled back to 1.1.13 on my x20r from a buggy 1.1.14 and had to find that out my self,good job it was at home before flying and not at the field...The rf side is serious and I have got a bit twitchy now on future updates as well as others have...EW

What did you find out? Failsafe icm with the older V2 on the RX should still work (although reported as not reliable...). I am about to rollback to 1.1.13 also because of the telemetry link issues with 1.1.14

**Comment by 4qr1hu9n7** [comment]
> Agreed,we are grateful for the development and passion for improvement on ETHOS but I am slightly confused over this V3 release because it says it corrects(in conjunction with 1.6.0 i understand) the failsafe issues by saving the data to the rx for permanent storage and not just waiting for 10/15 sec to receive this data every time we switch on.... So the point is ,as I see it..if this is a failsafe correction,and its not mandatory then that's a total contradiction in terms. If the failsafe issue is bad enough for a re write then surely it should be mandatory,and in which case if i choose not to update to v3 and ETHOS 1.6.0 I am missing out on a safety update. I don't want to update all my rx and tx ,just for a small failsafe issue.If the failsafe issue is solved in 1.6.0 ETHOS alone without the V3 rf upgrade,thats fine as its a developement of ETHOS.....I ,like many on here are trying to find a safe version and place to STOP any upgrades and enjoy flying but it never comes....A lot of the fixes i see i would need for a stable ETHOS are in 1.6.0 so its totally frustrating. I would love all the fixes to be addressed in one release(like futaba would do) so i get a bug free ETHOS on which to STOP and fly with for the next 15yrs if i don't need any more Features. There are bug fixes in the future releases so I cant stop now as i might hit on a bug in the future if trying something different and all the requests for new features will create more bugs....Its endless and frustrating...I am sold on Frsky and love the radios but crave a safe and stable setup for my planes so I can enjoy flying one again with no issues..EW

My sentiments exactly!! Stop with so many updates that address previous version fixes!! Get it right the first time!!

**Comment by robthomson** [comment]
Life does not work like that mate.   If it did.. We would all live in a utopia with perfect leaders.. No corruption etc etc.

I think you forget how incrediblly complex this stuff is.  And how all of these updates are being given to you for nothing to improve and make things better.

I hardly see any other brands with the same level of commitment to the product and community.

So let's not moan and put down the efforts.  Maybe let's offer help with testing. (which is what the beta fw are for!!!!!)

**Comment by Dsrenger** [comment]
Thanks rob

**Comment by 4qr1hu9n7** [comment]
> Life does not work like that mate. If it did.. We would all live in a utopia with perfect leaders.. No corruption etc etc.
> 
> I think you forget how incrediblly complex this stuff is. And how all of these updates are being given to you for nothing to improve and make things better.
> 
> I hardly see any other brands with the same level of commitment to the product and community.
> 
> So let's not moan and put down the efforts. Maybe let's offer help with testing. (which is what the beta fw are for!!!!!)

Okay. Then you can be the beta tester. Let's see how many airplanes you wreck after installing highly questionable updates!!

**Comment by robthomson** [comment]
With all due respect.

You are commenting on a beta firmware 3. 0 release

If you are using a beta flight fw (1.6) and a beta rf firmware.. You should know the risks.

If you do not want to take the risk and contribute.  Then move on.

That's absolutely fine.   

Testing these takes time and effort,  lots of feedback.  Sometimes there are bugs.  That is what testing is for. 

Please do not use this as an opportunity to troll in the background.  It's not productive and really does not help solve issues.

**Comment by bsongis** [comment]
@ETHOSWarrier There is something I don't understand, since Futaba does everything right (no bugs, never a problem with unique ID, a perfect Lua interpreter to give the users the maximum of flexibility, should I continue?) ... why do you continue to use FrSky gear?

**Comment by bsongis** [comment]
@all moaning people ...
If you don't need the RF new features (which I have listed), you don't have to install the RF update.
If you don't need the Ethos 1.6.0 new features (listed too), you don't have to install Ethos 1.6.0

**Comment by Alex00101** [comment]
Let me please attempt to provide some clarity (at least in my humble opinion):
1. The issue is not with Ethos, clearly. This should (and will be) raised in the RF_Feedback group. I think people are getting up in arms in here because there are more users in Ethos group then elsewhere. When it comes to upgrade process, Ethos has it ironed out (even when at times Lua changes require script updates) and responsiveness and quality of Ethos updates and feedback is one of the key advantages of Ethos. Pleading to everyone to keep this in mind when getting "emotional".
2. The issue - its root cause at least - is also not with the necessity of the update. RF updates including the last few do introduce important safety features and functions, as well as critical bug fixes, so I;d imagine for most "update or not update" is not a question. I'm guessing here, yes, however the fact is that in all my dealings with FrSky support 1st question asked is "are you on the latest firmware". So really "do not update if you don't have to" is not working. So yes RF updates are annoying but inevitable. 
3. The real issue is an apparent trend that each RF update (at least to Stab receivers) requires **reset, re-calibrate, re-setup**. On top of that, OTA, being described as on of key ACCESS selling points, clearly cannot be used safely. I bricked one of my stabs when applying the last update (the one when V3 scripts were introduced) OTA, and the 1st question support was asking "were you updating OTA? Yes? Hmmm....". Plus I have seen multiple complaints about OTA reliability. So the tendency is that with each RF update I need to physically access/run a wire to all updated receivers, when its a Stab I also have to get it out of the model, re-calibrate, install back in the model and do Voodoo dance of a very complicated  (tick this, untick this, unreverse that, reverse that) set up? Had to do it few months ago, now have (please see point 2 on why I use word "have") to do it all over again... what about the next RF update? Imagine if Ethos upgrades forced users to reprogram all models?

Bottom line:
- This is RF firmware not Ethos issue.
- "Do not upgrade" is not the answer, eventually it has to be done.
- I'd like to ask FrSky to develop a version of RX firmware that allows export of calibration data (and preferably stab setup data) before the upgrade and import back to the RX after the update. Preferably with more reliable OTA. I'd also ask this to be included into the upcoming major release (the same one we are talking about) so this should be the last time in a long while when full reset, re-calibrate, re-setup is required. Personally I'm happy to wait.

**Comment by ETHOSWarrier** [comment]
> @ETHOSWarrier There is something I don't understand, since Futaba does everything right (no bugs, never a problem with unique ID, a perfect Lua interpreter to give the users the maximum of flexibility, should I continue?) ... why do you continue to use FrSky gear?

Bsongis,you ask why I continue to use FrSky gear, fair point, Its because I am a loyal FrSky user now ,and  was let down by Futaba despite being a loyal user of them  for over 35yrs.
I am not "a moaner" my txt contains thanks and recognition of your work but I do see some minor similarities  with Futaba and just want to draw you attention to it.

Here in the UK Futaba decided to stop their latest 16IZ transmitting FASST and  it was going to be my next radio as a loyal user to Futaba but we were royally let down.I had about 40  planes and hi-performance gliders on FASST but it was decided to no longer support it even though their high end £3k radios did FASST so it can't be regulations as they said it was.

So now I have a system that I have no progression from and thus, as the TX  was 15yrs old and wearing out  I sold it all and bought into  the FrSky way....and I am pleased with FrSky  hardware and what it can do.Telemetry is years ahead of Futaba and obviously the software is good to but does leave a lot of experimentation to the user.

Now to address my point, even you have contradicted the issue by saying we don't have to go to 1.6.0 or rf V3......Thats not the impression i was given by some of your Frsky helpers on here....I had issues on rf 1.1.14 with my x20R and rolled back to 1.1.13 to solve it myself and your helpers mentioned V3 as the solution to the RF issues.This changes the direction of the discussion somewhat as we are being told one thing ,then its contradicted later ...so where do we ask and what do we do about it?? 

On Here of course and I don't expect to be called a moaner.This is , as a flyer, a serious business and people can die if my 4000watt moulded glider loses its rf link. As I have said before on other threads, I am not trolling, I don't know what that even is,I am trying to give you good info about how my radio is performing to help you solve issues and make it safe for others..

I have had to roll back both my x20s to 2.2.6 and my x20R to 1.1.13 to solve RSSI issues and Sensor ID issues...but am told future V3 rf updates will make this right...so which is it,,All i said was I don't feel totally at ease as you list the Bugs currently ,and their solutons in ETHOS 1.6.0 etc.... so I still cant relax as there are known bugs...but you say  I don't have to upgrade to 1.6.0 or V3 for RF issues ...This does not make sense.

I class my self as a Loyal  FrSky user now ,I have an x18,x18se x20s and an x20R...so I have contributed quite a bit to the financial success of frsky and currently have more than 40 Rx as well.I just don't want FrSky to lose focus on its users and we are not beta testers...you label the ETHOS releases as final..I load them...you label the rf releases as final.....I load them on to my radio.....how loyal is that..I am told V3 will cure future RF issues in conjunction with ETHOS 1.6.0 ..fine ..I will load it when labelled as final too so I am not being ungrateful just because i mention on here major rf issues.

My reference to Futaba's releases was because they labelled them ready ,so I loaded them too on my futaba radio but were few and far between due to their testing first. But due to their probable over cautious approach their development of software got years behind FrSky ,and after being let down I have fully taken on board your way of working.I am on your side Bsongis ,just don't leave me with 40 unusable rx again!  ...EW :)


Ps.....All i am trying to say is can we have more focus on solving the Bugs first (like Futaba) then we don't need to upgrade at all if we don't need the new features. I am happy as it is now for features, as are many of us,  but will have to upgrade the software/firmware to solve the bugs, but with new features added, instead/aswell as bug fixes, these cause more bugs and the cycle goes on for ever...EW

PPS...So for example ..can we have an  ethos 1.5.50 all current listed bugs fixed..end of....and an rf v2.50 all current  bugs fixed then we can fly for the next 15yrs or so relaxed knowing we are not going to hit any other issues which could cause rf lockout or the like and you wont hear a squeak out of me again :)

**Comment by bsongis** [comment]
It's difficult for me to answer. I am not aware of everything on the RF side of things as I am only in charge of Ethos and don't say one bad word about Ethos 1.5.0 or even Ethos 1.6.0, which is clearly labeled as beta.

We try to have as many non-regression tests to ensure the stability of the system. But believe me it's not easy to introduce new features without breaking anything, there is always a function that the users don't use the way you have in mind!

**Comment by 4qr1hu9n7** [comment]
Update. I chose to remove "All" of my TD SR18 receivers. Big job. Before doing so.I wanted to see what happens when the TD SR18 firmware is "not" updated. These are my observations. All of my models that have the TD SR18 installed worked. The model responded to control inputs and there were no errors in the model program. However! The TD SR18 receivers "DID NOT TRANSMITT ANY TELEMETRY" Don't know if this is a big deal or not. I then removed the receivers, updated the firmware and performed all procedures recommended. Okay. No issue. Very easy. I like to new calibration menu in the Ethos 1.6.0. Simple to follow. Anyway. Re installed all of the receivers.  Re bound the receiver. Everything worked as before the update plus all telemetry was restored. As for all of my other legacy receivers are concerned. There were absolutely no issues in any of the receivers. Made sure that the firmware on all of the legacy receivers was up to date. I just re bound all the receivers and set and tested my failsafe. I didn't want to go to this length to update my equipment nor do I want to go through this again. There must be a better way!!!
